src/internal/Modal.d.ts

```diff
@@ -1,10 +1,14 @@
 import * as React from 'react';
-import { StyledComponent } from '..';
+import { StandardProps } from '..';
+import { BackdropProps } from './Backdrop';
 import { TransitionDuration, TransitionHandlers } from './transition';
 
-export type ModalProps = {
+export interface ModalProps extends StandardProps<
+  React.HtmlHTMLAttributes<HTMLDivElement> & Partial<TransitionHandlers>,
+  ModalClassKey
+> {
   BackdropClassName?: string;
-  BackdropComponent?: React.ReactType;
+  BackdropComponent?: React.ComponentType<BackdropProps>;
```

Comment:
Restoring this to the type with more information; it seems like it was removed by mistake?

Reply:
Or was it changed to allow for a `string`? Should it be

```ts
BackdropComponent?: string | React.ComponentType<BackdropProps>;
```

?

Reply:
Yes, we allow a string to be provided. Also, I don't see what we should restrict ourselves to `BackdropProps`, unless I don't understand what `React.ComponentType<BackdropProps>` mean.

Reply:
@oliviertassinari the way that `BackdropComponent` is used inside `Modal.js` is:

```ts
<BackdropComponent
  invisible={BackdropInvisible}
  className={BackdropClassName}
  onClick={this.handleBackdropClick}
/>
```

Doesn't that mean that you should be providing a component which takes only those props? That's what `React.ComponentType<BackdropProps>` means.

Reply:
@pelotom I'm wondering. Is extra degree of freedom in this situation can harm? Either way, I'm happy with both typings.